### PR TITLE
[#12120] Instructor help page: page not scrolling to content when anchor links used

### LINKED ISSUE #12120: Instructor help page: page not scrolling to content when anchor links used
- **Environment**: `V8.23.0`

**Steps to reproduce**
Visit https://teammatesv4.appspot.com/web/front/help/instructor?section=questions&questionId=essay-questions, or links to any other section in instructor help page.

**Expected behaviour**
Scrolls to section.

**Actual behaviour**

Does not scroll to section.

https://user-images.githubusercontent.com/60355570/220666088-13bbf85c-28ba-4c8b-bae2-b598696ece2d.mov

### PR DESCRIPTION
**Pull request by SetsunaS**
Fixes #12120

**Outline of Solution**
First of all, I made changes to the `ngAfterViewInit()` function to accommodate the `section` and `questionId` anchors:
- I introduced a `target` variable, which identifies the more precise anchor between `section` and `questionId`
- if the anchor has a `questionId`, the corresponding question tab is expanded directly, and not in `scrollTo()`, because we don't need to do this if we only have `section` in the URL parameters
- the call to `scrollTo()` is also modified to handle the case where we only have `section` in the URL parameters, without a `questionId`

I also changed the `scrollOffset` in `scrollTo()` to 60, because I think it feels better. However, feel free to undo this change if you prefer the previous value.

Finally, I added the changes related to bug #12222 to ensure that we scroll correctly to the appropriate anchor.

### CODE REVIEW

**Review by zhaojj2209 — APPROVED** [review]
Thanks for the fix!

Do fix the merge conflict before we can merge your PR.

**Comment by SetsunaS** [comment]
The conflict has been resolved, thanks for your approval!

**Review by weiquu — APPROVED** [review]
LGTM!

**Pull request merged**
